### PR TITLE
Don't accept input until slash commands are loaded

### LINKED ISSUE #11145: Predictable /slash command loading for integration tests
### What happened?

Currently, slash commands are loaded async to the application start. This means, a user is able to start interacting with the system prior to all commands being loaded. 

This introduces many edge cases that span from breaking user experiences to "flakey" integration tests. 

This issue focuses on ensuring that all slash commands are completely loaded before allowing user interaction (ideally within the initializing UI)

Additionally, we also reload these twice to wait for the config to instantiate. Maybe we should just wait for the config to finish instantiating.

### What did you expect to happen?

test should be able to tell when the slash commands re loaded.

### Client information

<details>
<summary>Client Information</summary>

Run `gemini` to enter the interactive CLI, then run the `/about` command.

```console
> /about
# paste output here
```

</details>

### Login information

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by scidomino**
## TLDR

Wait for the initial slash commands to load before allowing text input. Also, show "Initializing..." until initial slash commands are available.

## Dive Deeper

This will make tests that depend on slash commands less flakey because they can be guaranteed that if the "Type your message" text is visible, that the slash commands are indeed loaded.

## Reviewer Test Plan

This is hard to test since the slash commands don't actually take very long to load. You can add code to make it wait 5 seconds in `CommandService.create()` and then you can see the behavior when starting up.

## Linked issues / bugs

Fixes #11145

### CODE REVIEW

File: packages/cli/src/ui/hooks/slashCommandProcessor.ts [hunk 1]
```diff
@@ -257,20 +259,18 @@ export const useSlashCommandProcessor = (
...
+          new BuiltinCommandLoader(config),
+          new FileCommandLoader(config),
+        ],
         controller.signal,
       );
       setCommands(commandService.getCommands());
-    };
-
-    load();
+    })();
```
[the remark this answers is not in the document]
Comment (scidomino) [reply packages/cli/src/ui/hooks/slashCommandProcessor.ts:273]: No. If this throws an error, the whole program will break. Best to let it throw.

**Review by bdmorgan — APPROVED** [review]
